samples/Matty/Matty.Client/Program.cs

```diff
@@ -0,0 +1,115 @@
+using OpenIddict.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Matty.Client
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            Console.WriteLine("+-------------------------------------------------------------------+");
+            Console.WriteLine("             Welcome to Matty a device code flow sample");
+            Console.WriteLine("+-------------------------------------------------------------------+");
+            Console.Write(" - Please press Enter after the server started...");
+            Console.ReadLine();
+
+            using var client = new HttpClient();
+
+            try
+            {
+                // Make call to "/.well-known/openid-configuration" endpoint
+                // to discover identity server configuration document
+                var discovery = await client.GetDiscoveryDocumentAsync("https://localhost:44321");
+                if (discovery.IsError)
+                    throw new Exception(discovery.Error);
+
+                var deviceResponse = await client.RequestDeviceAuthorizationAsync(new DeviceAuthorizationRequest
+                { 
+                    Address = discovery.DeviceAuthorizationEndpoint,
+                    Scope = "openid offline_access profile email",
+                    ClientId = "device",
+                });
+                if (deviceResponse.IsError)
+                    throw new Exception(deviceResponse.Error);
+
+                Console.WriteLine("+-------------------------------------------------------------------+");
+                Console.WriteLine("Device endpoint response");
+                Console.WriteLine($" - Device code: {deviceResponse.DeviceCode}");
+                Console.WriteLine($" - User code: {deviceResponse.UserCode}");
+                Console.WriteLine($" - Verification uri: {deviceResponse.VerificationUri}");
+
+                Console.WriteLine("+-------------------------------------------------------------------+");
+                Console.WriteLine(" - Please open the verification uri in the browser");
+                Console.WriteLine(" and enter the user code to authorize this device ");
+                Console.Write(" - Please press Enter to proceed...");
+                Console.ReadLine();
+
+                TokenResponse tokenResponse;
+                do
+                {
+                    tokenResponse = await client.RequestDeviceTokenAsync(new DeviceTokenRequest
+                    {
+                        Address = discovery.TokenEndpoint,
+                        ClientId = "device",
+                        DeviceCode = deviceResponse.DeviceCode
+                    });
+
+                    if (tokenResponse.IsError && tokenResponse.Error == "authorization_pending")
```

Comment:
If you want to avoid the hardcoded string, there's a constant in OpenIddict's `Error` class:

```csharp
if (tokenResponse is { IsError: true, Error: Errors.AuthorizationPending })
```